lib/util/http/github.ts

```diff
@@ -273,12 +278,33 @@ export class GithubHttp extends Http<GithubHttpOptions, GithubHttpOptions> {
     options?: InternalHttpOptions & GithubHttpOptions,
     okToRetry = true
   ): Promise<HttpResponse<T>> {
-    const opts = {
+    const opts: GithubHttpOptions = {
       baseUrl,
       ...options,
       throwHttpErrors: true,
     };
 
+    if (!opts.token) {
+      const authUrl = new URL(resolveBaseUrl(opts.baseUrl!, url));
+
+      if (opts.repository) {
+        // set authUrl to https://api.github.com/repos/org/repo or https://gihub.domain.com/api/v3/repos/org/repo
+        authUrl.hash = '';
+        authUrl.search = '';
+        authUrl.pathname = joinUrlParts(
+          authUrl.pathname.startsWith('/api/v3') ? '/api/v3' : '',
+          'repos',
+          `${opts.repository}`
+        );
+      }
+
+      const { token } = findMatchingRules(
+        { hostType: this.hostType },
+        authUrl.toString()
+      );
+      opts.token = token;
```

Comment:
And this means that for the case of github endpoints, we will bypass the generic http auth/hostRules logic in every case?

Reply:
yes

Reply:
Have we refactored all our tag/release fetching to use GraphQL now? i.e. we no longer face the pagination problem where `/repos/a/b/tags?page=1` ends up getting a link header pointing to `/repo/12345/tags?page=2`?

Reply:
yes, this forces to use same token, even the paged like points to the repo id.

Reply:
i can add an additional tests case if you think it makes it more safe